azure/datalake/store/core.py

```diff
@@ -748,6 +748,13 @@ def _put_data(rest, op, path, data, max_attempts=10, **kwargs):
         except (PermissionError, FileNotFoundError) as e:
             rest.log_response_and_raise(resp, e)
         except Exception as e:
+            if '"BadOffsetException"' in repr(e):
```

Comment:
Again, this isn't quite right. In our Multithreaded upload implementation, we have reasonable confidence that we will always be either at the right offset (end of file) or behind the end of file (append already succeeded). But in the core logic, someone else could use our "append" implementation and incorrectly try to append past the end of the file. That will also result in a BadOffsetException, but if they retry we will silently continue, even though their data is not getting committed (since it is beyond the end of the file).

The right place to do this logic is inside of our Multithread.py implementation, where we have confidence in our offset management and can be explicit about how to handle badoffsetexceptions. We should not make a general statement of how badoffsetexceptions should be handled for all custom clients that use our core.py, since that can be surprising behavior.